### PR TITLE
ADD: ImageRecognition

### PR DESCRIPTION
**Pull request by sihrc**

### CODE REVIEW

File: indicoio/images/recognition.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from indicoio.utils.image import image_preprocess
+from indicoio.utils.api import api_handler
+
+def image_recognition(image, cloud=None, batch=False, api_key=None, version=None, **kwargs):
+    """
+    Given an input image, returns a dictionary of image classifications with associated scores
+
+    * Input can be either grayscale or rgb color and should either be a numpy array or nested list format.
+    * Input data should be either uint8 0-255 range values or floating point between 0 and 1.
+    * Large images (i.e. 1024x768+) are much bigger than needed, resizing will be done internally to 64x64 if needed.
```
Comment (madisonmay) [inline indicoio/images/recognition.py]: I think we should be resizing to (128, 128) internally?

Comment (madisonmay) [inline indicoio/images/recognition.py]: Nevermind, looks like it's a min_resize to 144

File: indicoio/images/recognition.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    .. code-block:: python
+
+       >>> from indicoio import image_recognition
+       >>> features = image_recognition(<filename>)
+
+    :param image: The image to be analyzed.
+    :type image: str
+    :rtype: dict containing classifications
+    """
+    image = image_preprocess(image, batch=batch)
```
Comment (madisonmay) [inline indicoio/images/recognition.py]: Pass in a size and min_axis=True?

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+def load_image(self, relpath, as_grey=False):
+    im = Image.open(os.path.normpath(os.path.join(DIR, relpath))).convert('L');
+    pixels = list(im.getdata())
+    width, height = im.size
+    pixels = [pixels[i * width:(i + 1) * width] for i in xrange(height)]
```
Comment (anneCarlson) [inline tests/__init__.py]: It's weird to me that `pixels` is changing dimensions, going from list to matrix. Perhaps renaming like so would be more clear:
```
pixel_list = list(im.getdata())
width, height = im.size
pixel_matrix = [pixel_list[i * width:(i + 1) * width] for i in xrange(height)]
return pixel_matrix
```

Comment (sihrc) [inline tests/__init__.py]: Ah we don't even need this function any more. was just moving some things around.

Comment (anneCarlson) [inline tests/__init__.py]: Even better!

**Pull request merged**